src/chapter1/ch1-q1.js

```diff
@@ -44,3 +44,25 @@ export function hasUniqueCharactersSort(str) {
   }
   return true;
 }
+
+/**
+ * An alternate solution. Possibly more concise. Will work
+ * whether passed in as a char array or as a plain string.
+ * 
+ * Time: O(N)
+ * Additional space: O(N)
+ *
+ * @param  {string} str   String to check, passed in as a string or character array
+ * @return {boolean}      True if unique characters, otherwise false
+ */
+function unique(str){
+  let usedChars = [];
+  for(const char of str){
+     const exists = usedChars.indexOf(char);
```

Comment:
Using `indexOf` inside another loop would increase the time complexity of this algorithm, so it would not longer be `O(n)`.

https://stackoverflow.com/questions/3562919/what-is-the-cost-complexity-of-a-string-indexof-function-call/3562939